### DISCUSSION TITLE
New Dependencies when upgrading to lastest version..

### DISCUSSION THREAD

**Discussion by artnose** [ImageMagick/ImageMagick#3008] [General]
Trying to update to the latest version of IM (7.0.10-27 to 7.0.10-51) on centos 7 with rpms.   I have EPEL repo installed..
  
I'm getting failed dependencies that can't be resolved w/ yum 
error: Failed dependencies:
        libHalf.so.12()(64bit) is needed by ImageMagick-libs-7.0.10-51.x86_64
        libIex-2_2.so.12()(64bit) is needed by ImageMagick-libs-7.0.10-51.x86_64
        libIexMath-2_2.so.12()(64bit) is needed by ImageMagick-libs-7.0.10-51.x86_64
        libIlmImf-2_2.so.22()(64bit) is needed by ImageMagick-libs-7.0.10-51.x86_64
        libIlmThread-2_2.so.12()(64bit) is needed by ImageMagick-libs-7.0.10-51.x86_64
        libImath-2_2.so.12()(64bit) is needed by ImageMagick-libs-7.0.10-51.x86_64
        libpng16.so.16()(64bit) is needed by ImageMagick-libs-7.0.10-51.x86_64
        libpng16.so.16(PNG16_0)(64bit) is needed by ImageMagick-libs-7.0.10-51.x86_64
        libraw_r.so.19()(64bit) is needed by ImageMagick-libs-7.0.10-51.x86_64

Can someone tell me what's this related to, and what's the best way to resolve this. 
I certainly don't like forcing the update, especially as it seems to relate to png support

y

**Comment by urban-warrior** [comment]
We build the ImageMagick binary RPMS under CentOS 8.  For CentOS 7, you may need to build from the ImageMagick source RPM, ImageMagick-7.0.10-51.src.rpm.  A better upgrade path might be https://helostore.com/blog/install-imagemagick-on-centos-7.

**Reply by artnose** [reply]
CentOS 8 is End of Life, which is a whole other story.. 

The imagemagick.org site says the binaries are for Centos 7.

The remi repo only has ImageMagick-6.9.11.

Was hoping to avoid building from scratch, because that has even more dependencies.   But I guess I'll try that.  Thanks!

**Comment by urban-warrior** [comment]
Fixed the web pages.  Remi lists ImageMagick-7 for at least CentOS 8.  Not sure for 7.  EOL of 8 is a whole year away :-).  You can always install from source:

```
$ sudo yum install ImageMagick-devel
$ git clone git@github.com:ImageMagick/ImageMagick.git ImageMagick-7.0.10
$ cd ImageMagick-7.0.10
$ ./configure --with-modules
$ make
$ make check
$ sudo make install
```

**Reply by artnose** [reply]
I wish it were that easy.   Step 1 don't work, conflicts w/ existing install, and is older version.  Step don't work ether.

**Reply by urban-warrior** [reply]
Step 1 is to ensure you have all the development libraries in place so the ImageMagick build can leverage many of the delegate libraries.  For step 2, try this command:
```
git https://github.com/ImageMagick/ImageMagick.git ImageMagick-7.0.10
```
